app/src/main/jni/native_camera_session.cpp

```diff
@@ -136,6 +137,41 @@ class NativeCameraSessionImplementation : public NativeCameraSession {
         // Prepare request for texture target
         ACameraDevice_createCaptureRequest(cameraDevice, TEMPLATE_PREVIEW, &request);
 
+        // Find highest supported FPS and use it in camera request
+        ACameraMetadata *cameraMetadata = nullptr;
+        camera_status_t camera_status = ACameraManager_getCameraCharacteristics(cameraManager,
+                                                                                cameraId.c_str(),
+                                                                                &cameraMetadata);
+        if (camera_status != ACAMERA_OK) {
+            log_error("Failed to load camera characteristics %d", camera_status);
+        } else {
+            ACameraMetadata_const_entry supportedFpsRanges;
+            ACameraMetadata_getConstEntry(cameraMetadata,
+                                          ACAMERA_CONTROL_AE_AVAILABLE_TARGET_FPS_RANGES,
+                                          &supportedFpsRanges);
+            int32_t highestRange[2] = {0, 0};
+            for (int32_t i = 0; i < supportedFpsRanges.count; i += 2) {
+                int32_t min = supportedFpsRanges.data.i32[i];
+                int32_t max = supportedFpsRanges.data.i32[i + 1];
+                log_debug("Supported camera FPS range: [%d-%d]", min, max);
+                // TODO: Respect targetFps here?
```

Comment:
It would indeed be good to somehow respect the target FPS, similarly to how the resolution and camera selection works. Just checked that at least Pixel 2 supports 60 FPS and only supporting the "max" setting is a bit problematic if it varies a lot between phones